src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileAction.java

```diff
@@ -1476,21 +1481,25 @@ protected Spawn createSpawn(Map<String, String> clientEnv) throws ActionExecutio
        * in-memory. We can do that via
        * {@link ExecutionRequirements.REMOTE_EXECUTION_INLINE_OUTPUTS}.
        */
-      executionInfo =
-          ImmutableMap.<String, String>builderWithExpectedSize(executionInfo.size() + 1)
-              .putAll(executionInfo)
-              .put(
-                  ExecutionRequirements.REMOTE_EXECUTION_INLINE_OUTPUTS,
-                  getDotdFile().getExecPathString())
-              .build();
+      executionInfo.put(ExecutionRequirements.REMOTE_EXECUTION_INLINE_OUTPUTS,
+          getDotdFile().getExecPathString());
+    }
+
+    if (shouldParseShowIncludes()) {
```

Comment:
nit: why not simply `booleaan shouldParseShowIncludes = featureConfiguration.isEnabled(CppRuleClasses.PARSE_SHOWINCLUDES`?

Every call site of `shouldParseShowIncludes()` is called from this single method, so it's a slightly better because then you have one less method and a teeny tiny bit less computation.

Reply:
Since `featureConfiguration` is null when running some unit tests. By using `shouldParseShowIncludes`, we open a chance for tests to mock this method.

Reply:
...and by doing so, widen the interface of `CppCompileAction`. If that's the motivation to extract this into a separate method, let's not do that and provide a `FeatureConfiguration` to the tests, if need be.